### DISCUSSION TITLE
pwsh - Set-PSReadLineOption -PromptText

### DISCUSSION THREAD

**Discussion by mattcargile** [JanDeDobbeleer/oh-my-posh#1641] [Troubleshoot] [answered]
I've been exploring this feature and I can't come to an easy conclusion. [This ](https://github.com/PowerShell/PSReadLine/issues/1997#issuecomment-736090172)is the nature of the feature. You can pass in a two item array for your ending `PromptText` where the first item is without a current error in your command line and the second item would be if there was a syntax error in the line ( or lines ) on the command line. Right now, my prompt doesn't immediately tell me if there is a syntax error visually.

I'm conflicted as my `PromptText` changes with failures of the previous command. I am using the [JanDeDobbeleer theme](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/jandedobbeleer.omp.json) more-or-less ( i.e. I've added to it ).

I think what I would like to do is to be able to `Set-PSReadlineOption -PromptText` depending on the `$?` and `$LASTEXITCODE`. It would be cool if I could retain that information in the `Set-PoshContext` `function` that is meant to be [aliased ](https://ohmyposh.dev/docs/text)with the user's version.

Anyone else solved this problem or should I just copy the code in the current build for the `Function:\prompt` and put it in my `Set-PoshContext`? I could also pick a color that is neither of my two possible colors like maybe orange.

**Comment by JanDeDobbeleer** [comment, marked as answer]
You just gave me the source to a missing piece 🤗 I'll have a look at how to implement this.

**Reply by mattcargile** [reply]
Awesome! Love it! Thanks, Jan!

**Reply by JanDeDobbeleer** [reply]
Just give me a week or so. I'm in the middle of a crazy refactoring so 😅

**Reply by mattcargile** [reply]
Sounds good. Want me to convert it into a more formal Issue/Feature Request?

**Reply by mattcargile** [reply]
I was playing with it a little to get it to work. It wouldn't work when only specifying the error text ( second item in the array ). I was trying to avoid the non consistent color when the last command was an error.

```powershell
Set-PSReadLineOption -PromptText "$($PSStyle.Foreground.FromRgb('#00897b'))`u{E0B4} $($PSStyle.Reset)","$($PSStyle.Foreground.BrightRed)`u{E0B4} $($PSStyle.Reset)"
```

**Reply by JanDeDobbeleer** [reply]
This is on my list as the second item to look at. If we can make it happen, we make it happen.

**Reply by JanDeDobbeleer** [reply]
It's live (or at least, it's on its way to be there)

**Reply by mattcargile** [reply]
Appreciate it! I see the Continuation Prompt too. 

I see for the `PromptText` it is a one time load. How should I handle it when I have an error in the last command and the prompt changes color? Is there a way to have different colors using the `template` logic?

**Reply by mattcargile** [reply]
I see how we can't see the information derived in the `function:prompt` like the consolidated exit code or the completion time in the jandedobbeleer theme?

**Reply by mattcargile** [reply]
I think I like how you've got it. I just overlay the initial prompt segment with your checkmark and exclamation part where there was an error in the previous line or not.

**Reply by mattcargile** [reply]
Sorry to keep throwing out my thoughts in small messages. I see `{{ .Code}}` and `{{ .Ms }}` wouldn't be available as it isn't global and I'd have to derive my own `$env` variables to be able to use that data in logic in the `transient_prompt` or the `secondary_prompt` or the `valid_line`/`error_line`.

**Reply by JanDeDobbeleer** [reply]
@mattcargile I can create globals for that so they're available if you want.

**Reply by JanDeDobbeleer** [reply]
@mattcargile I had a look and allowing these values would imply we need to render the secondary prompt on every prompt change. That's not something worth considering it can impact performance.
